### PR TITLE
🌱 Align v1beta3 types to latest changes in Kubernetes

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR alignes v1beta3 types to latest changes in Kubernetes

For sake consistency, some minor changes (+optional) are applied to KubeadmConfig in Cluster API v1alpha4 API.

@vincepri @CecileRobertMichon

### CODE REVIEW

File: bootstrap/kubeadm/types/v1beta3/types.go [hunk 1]
```diff
@@ -277,47 +324,62 @@ type JoinConfiguration struct {
...
 	// ControlPlane defines the additional control plane instance to be deployed on the joining node.
 	// If nil, no additional control plane instance will be deployed.
+	// +optional
 	ControlPlane *JoinControlPlane `json:"controlPlane,omitempty"`
+
+	// SkipPhases is a list of phases to skip during command execution.
+	// The list of phases can be obtained with the "kubeadm join --help" command.
+	// The flag "--skip-phases" takes precedence over this field.
+	// +optional
+	SkipPhases []string `json:"skipPhases,omitempty"`
```
Comment (CecileRobertMichon) [inline bootstrap/kubeadm/types/v1beta3/types.go:348]: I thought we were going to hold on adding SkipPhases for now?

Comment (fabriziopandini) [reply bootstrap/kubeadm/types/v1beta3/types.go:348]: We are adding  it only to the v1beta3 types, which are a mirror of the kubeadm ones used only when serialising the Cluster API's KubeadmConfig object.
However we are not adding skip phases to the v1alpha4 KubeadmConfig object, so this field will be hidden to the CAPI users

**Review by vincepri — APPROVED** [review]
/lgtm

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**